bitcoin/varint.h

```diff
@@ -14,4 +14,13 @@ size_t varint_put(u8 buf[VARINT_MAX_LEN], varint_t v);
 
 /* Returns bytes used: 0 if max_len too small. */
 size_t varint_get(const u8 *p, size_t max_len, varint_t *val);
+
+
+/* Big-endian variant of varint_put, used in lightning */
+size_t bigsize_put(u8 buf[VARINT_MAX_LEN], varint_t v);
+
+/* Big-endian variant of varint_get, used in lightning */
+size_t bigsize_get(const u8 *p, size_t max, varint_t *val);
```

Comment:
These routines are in the wrong place, and they're redundant.

You want `fromwire_bigsize()` and `towire_bigsize()`.

Reply:
Agreed that they don't belong here, but they are useful since `fromwire_*` /  `towire_*` require `tal_arr` which they can extend, whereas the sphinx operations require in-place operations. I tried using the `fromwire` and `towire` variants, but it turned out to involve way more work than just implementing it again.

I can move the functions in a later cleanup PR.